mkosi/machine.py

```diff
@@ -168,21 +178,19 @@ def __init_subclass__(cls, args: Sequence[str] = []) -> None:
 
     @classmethod
     def setUpClass(cls) -> None:
-        if os.getuid() != 0:
-            raise unittest.SkipTest("Must be invoked as root.")
-
-        cls.machine = Machine(cls.args)
+        with pytest_skip_not_supported(cls.args):
+            cls.machine = Machine(cls.args)
 
-        verb = cls.machine.args.verb
-        no_nspawn = parse_boolean(os.getenv("MKOSI_TEST_NO_NSPAWN", "0"))
-        no_qemu = parse_boolean(os.getenv("MKOSI_TEST_NO_QEMU", "0"))
+            verb = cls.machine.args.verb
+            no_nspawn = parse_boolean(os.getenv("MKOSI_TEST_NO_NSPAWN", "0"))
+            no_qemu = parse_boolean(os.getenv("MKOSI_TEST_NO_QEMU", "0"))
 
-        if no_nspawn and verb == Verb.boot:
-            raise unittest.SkipTest("Nspawn test skipped due to environment variable.")
-        if no_qemu and verb == Verb.qemu:
-            raise unittest.SkipTest("Qemu test skipped due to environment variable.")
+            if no_nspawn and verb == Verb.boot:
+                raise unittest.SkipTest("Nspawn test skipped due to environment variable.")
+            if no_qemu and verb == Verb.qemu:
+                raise unittest.SkipTest("Qemu test skipped due to environment variable.")
```

Comment:
For now we only need it at the __init__ method, since this PR only merges the CI tests.
Once we allow machines that can't be booted to be built there will be more usage of the `skip_not_supported` context manager.